libminifi/src/Configuration.cpp

```diff
@@ -179,12 +179,10 @@ std::vector<std::string> Configuration::getSensitiveProperties(const std::functi
 }
 
 bool Configuration::validatePropertyValue(const std::string& property_name, const std::string& property_value) {
-  for (const auto& config_property: Configuration::CONFIGURATION_PROPERTIES) {
-    if (config_property.name == property_name) {
-      return config_property.validator->validate(property_name, property_value).valid();
-    }
-  }
-  return true;
+  if (!Configuration::CONFIGURATION_PROPERTIES.contains(property_name))
+    return true;
+
+  return Configuration::CONFIGURATION_PROPERTIES.at(property_name)->validate(property_name, property_value).valid();
```

Comment:
We should do the lookup only once, not twice.
```suggestion
  const auto validator = Configuration::CONFIGURATION_PROPERTIES.find(property_name);
  if (validator == std::end(Configuration::CONFIGURATION_PROPERTIES))
    return true;

  return (*validator)->validate(property_name, property_value).valid();
```